src/Atmos/Parameterizations/CloudPhysics/Microphysics.jl

```diff
@@ -233,7 +233,8 @@ function conv_q_rai_to_q_vap(
     _MP_n_0::FT = MP_n_0(param_set)
 
     if qr > FT(0)
-        qv_sat = q_vap_saturation(param_set, T, ρ, q)
+        ts_neq = TemperatureSHumNonEquil(param_set, T, ρ, q)
+        qv_sat = q_vap_saturation(ts_neq)
         q_v = q.tot - q.liq - q.ice
```

Comment:
I just need saturation vapour specific humidity here. Why is it better to construct the whole thermo state to get just that one number?

Reply:
Because we need to change the interface of `q_vap_saturation(param_set, T, ρ, q)`, and we'd also like to continue being able to change the interface. The idea is that the interface of the thermodynamic states should be stable. Otherwise, this line will read

```julia
phase_type = MoistThermodynamics.PhaseNonEquilType() # internal to moist thermodynamics
qv_sat = q_vap_saturation(param_set, T, ρ, phase_type, q)
```

And this may further change when we add other thermodynamic states (for ocean, land, etc.)

Reply:
Is it optimised by the compiler? If not it's an overhead that will be done in every grid point every time I compute RHS.

Reply:
I don't think performance is an issue here. The non-equil states do not need an iterative solver so it's only a few extra flops (which will be hidden by MPI communication). The main advantage of leveraging the high-level interface is that it can be more extensively tested for correctness and consistency. There are fewer guarantees on correctness/consistency when using moist thermo's internals (since users will then need to provide these guarantees).